File: code-generator/src/main/java/io/github/potjerodekool/openapi/internal/di/SingletonScopeManager.java
package io.github.potjerodekool.openapi.internal.di;

import org.checkerframework.checker.nullness.qual.NonNull;

public class SingletonScopeManager<T> implements ScopeManager<T> {

    private final @NonNull T instance;

    public SingletonScopeManager(final @NonNull T instance) {
        this.instance = instance;
    }

    @Override
    public T get(final DefaultApplicationContext applicationContext) {
        return instance;
    }

    @Override
    public Class<T> getBeanType() {
        return (Class<T>) instance.getClass();
    }
}

File: code-generator/src/main/kotlin/io/github/potjerodekool/openapi/internal/di/bean/SingletonScopeManager.kt
package io.github.potjerodekool.openapi.internal.di.bean

class SingletonScopeManager<T>(private val instance: T) : ScopeManager<T> {
    override fun get(applicationContext: DefaultApplicationContext?): T {
        return instance
    }

    override val beanType: Class<T>
        get() = instance!!::class.java as Class<T>
}
